core/src/test/scala/BlockSpec.scala

```diff
@@ -107,44 +118,44 @@ class BlockSpec extends AnyFlatSpec {
   }
 
   it should "be able to make partition" in {
-    val keyStart1 = new Key(Array(0x0))
-    val keyEnd1 = new Key(Array(0x4))
-    val keyRange1 = new KeyRange(keyStart1, keyEnd1)
-    val workerMetadata1 = WorkerMetadata("1.1.1.1", 123, Option(keyRange1))
-
-    val keyStart2 = new Key(Array(0x5))
-    val keyEnd2 = new Key(Array(0x9))
-    val keyRange2 = new KeyRange(keyStart2, keyEnd2)
-    val workerMetadata2 = WorkerMetadata("2.2.2.2", 123, Option(keyRange2))
+    val firstKeyRange = KeyRange(new Key(Array(0x0)), new Key(Array(0x4)))
```

Comment:
In this case, 0x4 key cannot included in any partition. I wonder that it is intended
eg.`new Record(new Key(Array(0x4)), Array(0x1, 0x2, 0x3))` never included in KeyRange

Reply:
Intended. Many range interfaces DO NOT include it. e.g. python's `ragne`, scala's `Range`.

Reply:
778f9fb06c45faf6efcaad42f30a44dc22df31e6